civitai/civitai
# Post Intent

## what this does
- add intent post page
- remove orchestrator urls from transmitter, add stage to handledrop
- add max video size
- handle external meta data
- ability to add string tags to post
- add column "domain" for Tool table
- global post tag limit
- add tool on post image creation if metadata is present
- add UI display for external image meta

## Testing instructions:

Example link: https://dev.civitai.com/intent/post?mediaUrl=https://files.catbox.moe/h675id.png&detailsUrl=https://dev.civitai.com/api/test&title=testing%20post&description=hello&tags=whatever,eboy

Try the following:
- bad params (missing mediaUrl, url that isn't valid, etc)
- url that isnt media (google.com)
- url that is media but has cors issues
- url that is too big (500mb image)

After creating:
- details url whose domain is not allowed
- details url that doesnt work
- details url with bad params
- make sure tags are good, other things get set (title)
- check that valid details url was parsed successfully
- check external data with no image metadata, with image meta, no external, etc

For each:
- do image and video
